app/services/authentic_jobs_service.rb

```diff
@@ -0,0 +1,83 @@
+class AuthenticJobsService < JobFetcher
+  def self.scrape(term='ruby')
+    jobs = self.get_jobs(term)
+    self.format_and_create_entries(jobs, term)
+  end
+
+  def self.format_and_create_entries(entries, term)
+    entries.each do |entry|
+      formatted_entry = self.format_entry(entry, term)
+      self.create_records(formatted_entry) unless entry[:company].nil?
+    end
+  end
+
+  def self.format_entry(entry,term)
+    { job: {
+        title: self.full_title(entry),
+        url: entry[:url],
+        location: self.pull_location(entry),
+        raw_technologies: self.pull_technologies(entry[:description], term),
+        description: entry[:description],
+        remote: entry[:telecommuting] == 1 ? true : false,
+        posted_date: entry[:post_date]
+      },
+      company: {
+        name: self.pull_company_name(entry)
+      }
+    }
+  end
+
+  def self.full_title(entry)
+    if !entry[:company].nil?
+      "#{entry[:title]} at #{self.pull_company_name(entry)} (#{self.pull_location(entry)})"
+    else
+      entry[:title]
+    end
+  end
+
+  def self.pull_location(entry)
+    if !entry[:company].nil? && !entry[:company][:location].nil?
+      entry[:company][:location][:name]
+    elsif entry[:telecommuting] == 1
```

Comment:
You can write this conditional I think more succinctly:  entry.fetch(:company, {})[:location].nil?